vortex-io/src/object_store.rs

```diff
@@ -67,12 +67,23 @@ impl VortexReadAt for ObjectStoreReadAt {
         let buffer = match response.payload {
             GetResultPayload::File(file, _) => {
                 unsafe { buffer.set_len(len) };
-                tokio::task::spawn_blocking(move || {
-                    file.read_exact_at(&mut buffer, range.start)?;
-                    Ok::<_, io::Error>(buffer)
-                })
-                .await
-                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))??
+                #[cfg(feature = "tokio")]
+                {
+                    tokio::task::spawn_blocking(move || {
```

Comment:
Wait, why was this one even wrapped in a tokio task:

```
                    tokio::task::spawn_blocking(move || {
                        file.read_exact_at(&mut buffer, range.start)?;
                        Ok::<_, io::Error>(buffer)
                    })
                    .await
```


```                        
file.read_exact_at(&mut buffer, range.start)?;
```
wasn't async before either.

I would have expected:
```
file.read_exact_at(&mut buffer, range.start).await?;
```
@gatesn Probably has a hunch why.

Reply:
This is a blocking call, since its not tokio but, std file

Reply:
Yeah what I meant was that I would have expected that we use an async read call here.

Reply:
Along the lines of https://docs.rs/tokio/latest/tokio/fs/fn.read.html.

Reply:
I think this is a different PR no?